cocos/2d/CCTMXXMLParser.cpp

```diff
@@ -746,7 +746,7 @@ void TMXMapInfo::endElement(void *ctx, const char *name)
 
             uint32_t* bufferPtr = reinterpret_cast<uint32_t*>(buffer);
             for(auto gidToken : gidTokens) {
-                auto tileGid = (uint32_t)strtol(gidToken.c_str(), nullptr, 10);
```

Comment:
If value is bigger then "long int" strtol returns it can represent the maximum integer.
In win32 "long int" is same as "int".Value 0x80000001 in "strtol" will return 0x7FFFFFFF and "strtoul" will return 0x80000001